### ISSUE TITLE
pk414-standard crashes on startup when booting from a USB key

### ISSUE THREAD

**Issue by rustylynch** [clearlinux/distribution#43]
If you grab the latest live image, boot from it on a USB key, install the kernel-pk bundle, update the bootloader timeout so you can select the new kernel, and then reboot... then the boot will fail.

I am using a dell xps 13 laptop which doesn't have a serial port, so the best i can do is capture a picture of the kernel panic on the screen.  Normally there are multiple pages scrolling past quickly, making it impossible to see the first panic, but after adding a command line option to maxcpus=1 then i have a single clean panic about the inability to mount the rootfs.

![pk-crash-on-usb](https://user-images.githubusercontent.com/730249/38583095-c296a262-3cc6-11e8-8e44-d36637aef474.jpg)

**Comment by rustylynch** [comment]
Note that the exact same kernel works fine when booting off my hard drive.

**Comment by rustylynch** [comment]
Note that adding a rootwait to the kernel command-line works around this issue

**Comment by fenrus75** [comment]
We have a fix for this in our normal kernel... which apparently worked
great.
Any reason to not use the normal kernel on bare metal ?

**Comment by rustylynch** [comment]
I was attempting to exercise some hypervised use cases working off the tip of acrn and just wanted to ensure that the standard pk kernel was alive in kicking on bare metal first.

**Comment by rustylynch** [comment]
Note that applying the native kernel fix to the top of the patch list, with a quick rebase to remove the offset warning, the pk-standard works fine on bare metal usb without the need for an explicit rootwait

**Comment by iphutch** [comment]
@bwarden could you coordinate with Saul on this?

**Comment by bwarden** [comment]
@rustylynch Are we talking about 0125-init-wait-for-partition-and-retry-scan.patch?

**Comment by miguelinux** [comment]
@bwarden yes,

Send a PR to PK

**Comment by bwarden** [comment]
Do you think we should maintain it against linux-pk414 ourselves, or submit it to the upstream PK to be added to the master patch set?

**Comment by rustylynch** [comment]
@bwarden yes, 0125-init-wait-for-partition-and-retry-scan.patch is the patch i tested

**Comment by miguelinux** [comment]
@rustylynch 
Please try with 4.14.34-29 the patch is already integrated.
